docs/installation.rst

```diff
@@ -0,0 +1,103 @@
+will install both the core library and the command line interface.
+
+Installation with ``conda`` (recommended)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We recommend installing ``openfe`` with ``conda``, because it provides easy
+installation of other tools, including molecular dynamics tools such as
+OpenMM and ambertools, which are needed by ``openfe``. 
+Conda can be installed either with the `full Anaconda distribution
+<https://www.anaconda.com/products/individual>`_, or with
+the `smaller-footprint miniconda
+<https://docs.conda.io/en/latest/miniconda.html>`_. We also recommend
+`mamba
+<https://mamba.readthedocs.io/en/latest/installation.html#installation>`_ as
+a more perfomant variant of ``conda``.
+
+You can get ``openfe`` from the ``conda-forge`` channel. To install the most
+recent release, use the command ::
+
+  $ conda install -c conda-forge openfe
+
+With that, you should be ready to use ``openfe``!
+
+Single file installer
+~~~~~~~~~~~~~~~~~~~~~
+
+[COMING SOON!]
+
+.. TODO: maybe Mike can fill this in? just needs (1) how to download the
+   single file installer; (2) how to use the single file installer
+
+Developer install
+~~~~~~~~~~~~~~~~~
+
+If you're going to be developing for ``openfe``, you will want an
+installation where your changes to the code are immediately reflected in the
+functionality. This is called a "developer" or "editable" installation.
+
+Getting a developer installation for ``openfe`` first installing the
+requirements, and then creating the editable installation. We recommend
+doing that with ``conda`` using the following procedure:
+
+First, clone the ``openfe`` repository, and switch into its root directory::
+
+  $ git clone https://github.com/OpenFreeEnergy/openfe.git
+  $ cd openfe
+
+Next create a ``conda`` environment containing the requirements from the
+specification in that directory::
+
+  $ conda env create -f environment.yml
+
+Then activate the ``openfe`` environment with::
+
+  $ conda activate openfe
+
+Finally, create the editable installation::
+
+  $ python -m pip install -e .
+
+Note the ``.`` at the end of that command, which indicates the current
+directory.
+
+Optional dependencies
+---------------------
+
+Certain functionalities are only available if you also install other,
+optional packages.
+
+* **perses tools**: To use perses, you need to install perses and OpenEye,
+  and you need a valid OpenEye license. To install both packages, use::
+
+    $ conda install -c conda-forge -c openeye perses openeye
+
+Testing your installation
+-------------------------
+
+``openfe`` has a thorough test suite, and running the test suite is a good
+start to troubleshooting any installation problems. The test suite requires
+``pytest`` to run. You can install ``pytest`` with::
+
+  $ conda install -c conda-forge  pytest
+
+Then you can run the test suite (from any directory) with the command::
+
+  $ py.test --pyargs openfe openfecli
```

Comment:
isn't this `pytest` now? I thought the whole `py` prefix was being deprecated (you're showing your age :p)

Reply:
also maybe this would be a good thing to add to the cli... `openfecli validate_installation`

Reply:
I've had `alias pytest="py.test"` in my zshrc for so long that it was in my bashrc.

Not sure that I agree on the `validate_installation`, unless it does other things. In particular, not fond of running a subshell, and I don't offhand know how to run pytest from within a python process.

Reply:
I think it's just `pytest.main()` https://docs.pytest.org/en/7.1.x/how-to/usage.html#calling-pytest-from-python-code

I'm thinking it would be nice if there was a non-developer way to run the tests, which would pick up things like bad environments, linking problems etc.  Just a smoke test that is ran as part of the install process.